scipy/integrate/_quad_vec.py

```diff
@@ -122,6 +122,9 @@ def quad_vec(f, a, b, epsabs=1e-200, epsrel=1e-8, norm='2', cache_size=100e6, li
         Vector norm to use for error estimation.
     cache_size : int, optional
         Number of bytes to use for memoization.
+    limit : float or int, optional
+        An upper bound on the number of subintervals used in the adaptive
+        algorithm.
```

Comment:
Looking at the source code (as non-expert) this seems sensible given context/comments:

![image](https://user-images.githubusercontent.com/7903078/150656164-2e67a1d8-e3c5-4d15-b28a-045a972899ae.png)

In case we don't have an expert drop by on this one, I did a bit of digging on the other linked issue about why the default value may differ from `quad`. It seems that `quad` ultimately ends up calling quadpack Fortran routines like the one in `scipy/integrate/quadpack/dqagse.f`, which has an almost identical description for `limit`, and I see `F_INT limit=50;` in `scipy/integrate/__quadpack.h` for i.e., `*quadpack_qagse`.

Conversely, `quad_vec` appears to have a largely Python-based implementation. Perhaps the huge difference in backend implementation explains the vastly different `limit` defaults, but not sure.

This also makes me wonder--10,000 loops of Python doing pretty much anything, let alone integrating, could be slow. I wonder if there is some low-hanging fruit for improvement via Cython/pythran. Given 10k is the default, if the function is heavily used, it could make a difference. Anyway, that's out of scope here, and the new docstring render in CircleCI looks good too.

Reply:
Thanks, @tylerjereddy.

The single usage of this parameter in the `integrate.quad_vec` function occurs [here](https://github.com/scipy/scipy/blob/ecb800f37bc6c6eb4f160cba34fec81932eaea70/scipy/integrate/_quad_vec.py#L357).

My understanding is that, while `len(intervals) < limit`, the code:

1. Selects intervals with the largest errors for subdivision (see line 358).
2. Subdivides these intervals (increasing the value of `len(intervals)` -- see line 379).
3. Checks convergence conditions. Ideally this loop stops before reaching the `limit` parameter that we describe in the docs. (See line 390.)

If my understanding is correct, our description of the parameter `limit` should be both accurate and adequate.

Reply:
Concerning your comment on 10k lines of Python, I generally agree. I think the hope is that the convergence conditions after line 390 tend to be reached long before `len(intervals) >= limit`.